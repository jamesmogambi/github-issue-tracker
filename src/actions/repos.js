require("isomorphic-fetch");

// SET_REPOSITORIES
export const setRepositories = (payload) => ({
  type: "SET_REPOSITORIES",
  payload,
});
// CLEAR_REPOSITORIES
export const clearRepositories = () => ({
  type: "CLEAR_REPOSITORIES",
});

// ERROR
export const error = (payload) => ({
  type: "ERROR",
  payload,
});

// Get Github Repositories
export const getRepositories = (params) => {
  return (dispatch) => {
    dispatch(clearRepositories());
    fetch("https://api.github.com/graphql", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        authorization: `bearer ${process.env.GITHUB_ACCESS_TOKEN}`,
      },
      body: JSON.stringify({
        query: `
               query($github_login: String!) {
    user(login: $github_login) {
      name
      repositories(first: 100) { 
        nodes {
             id
        name
        description 
        url 
        updatedAt
        forkCount
        }
      }
    }
  }
                 `,
        variables: {
          github_login: params.username,
        },
      }),
    })
      .then((res) => res.json())
      .then((res) => {
        if (res.data.user) {
          let repos = res.data.user.repositories.nodes;
          let owner = res.data.user.name;
          dispatch(
            setRepositories({
              repositories: repos,
              repositoryOwner: params.username,
            })
          );
        }
      })
      .catch((err) => {
        dispatch(error(err));
        console.error(err);
      });
  };
};
