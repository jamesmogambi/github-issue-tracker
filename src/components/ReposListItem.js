import React from "react";
import { Link } from "react-router-dom";

const ReposListItem = ({ id, name, description, repositoryOwner }) => (
  <div className="list-item">
    <div> 
      <h3 className=" list-item__title">{name}</h3>
      <span className=" list-item__sub-title">{description}</span>
      <Link to={`/issues/${name}/${repositoryOwner}`}>
        <h3 className="list-item__sub-title list-link ">View Issues</h3>
      </Link>
    </div>
    <a className="button button--link " href={`https://github.com/${repositoryOwner}/${name}/tags`} target="_blank">Tags</a>
 </div>
);

export default ReposListItem;
