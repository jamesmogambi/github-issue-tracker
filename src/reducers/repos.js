// Repositories Reducer
const initialState = {
  repositories: [],
  repositoryOwner: null,
  error: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case "SET_REPOSITORIES":
      const { repositories, repositoryOwner } = action.payload;
      return {
        ...state,
        repositoryOwner,
        repositories,
      };
    case "CLEAR_REPOSITORIES":
      return {
        repositories: [],
        repositoryOwner: null,
        error: null,
      };

    default:
      return state;
  }
};
