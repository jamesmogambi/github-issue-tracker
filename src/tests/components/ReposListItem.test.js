import React from "react";
import { shallow } from "enzyme";
import repos from "../fixtures/repos";
import ReposListItem from "../../components/ReposListItem";

test("should render ReposListItem correctly", () => {
  const data = {
    repositoryOwner: "JamesEdward",
    ...repos[0],
  };
  const wrapper = shallow(<ReposListItem {...data} />);
  expect(wrapper).toMatchSnapshot();
});
