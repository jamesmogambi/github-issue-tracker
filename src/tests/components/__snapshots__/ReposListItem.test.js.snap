// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ReposListItem correctly 1`] = `
<div
  className="list-item"
>
  <div>
    <h3
      className=" list-item__title"
    >
      react-native-awesome-places-app
    </h3>
    <span
      className=" list-item__sub-title"
    >
      view awesome places.
    </span>
    <Link
      replace={false}
      to="/issues/react-native-awesome-places-app/JamesEdward"
    >
      <h3
        className="list-item__sub-title list-link "
      >
        View Issues
      </h3>
    </Link>
  </div>
</div>
`;
